packages/flutter_tools/test/general.shard/generate_localizations_test.dart

```diff
@@ -1909,6 +1909,46 @@ import 'output-localization-file_en.dart' deferred as output-localization-file_e
           )),
         );
       });
+
+      testWithoutContext('should throw attempting to generate a select message with an incorrect message', () {
```

Comment:
@asashour do you think it's worth doing a similar test for plurals ?

Reply:
Why not. However the plurals use a different mechanism as I thought before raising the PR.

Anyhow, test to be shortly added.

Reply:
A test case was added to ensure we don't `fail`. However I am not sure if we should throw an exception instead?

Reply:
I am not sure @asashour. For example. Intl.plural eventually has a check like this:

```
    ArgumentError.checkNotNull(other, 'other');
    ArgumentError.checkNotNull(howMany, 'howMany');
```

So we can assume that a invalid input like `{count, plural,}` really IS an error, except when it means that `other` is set to `''`. (I think this is unlikely but don't know).

For Intl.select _something_ has be to be defined, like shown in the code:

```
  /// Internal: Implements the logic for select - use [select] for
  /// normal messages.
  static T selectLogic<T>(Object choice, Map<Object, T> cases) {
    // This will work if choice is a string, or if it's e.g. an
    // enum and the map uses the enum values as choices.
    var exact = cases[choice];
    if (exact != null) return exact;
    // If it didn't match exactly, take the toString and
    // take the part after the period. We need to do this
    // because enums print as 'EnumType.enumName' and periods
    // aren't acceptable in ICU select choices.
    var stringChoice = '$choice'.split('.').last;
    var stringMatch = cases[stringChoice];
    if (stringMatch != null) return stringMatch;
    var other = cases['other'];
    if (other == null) {
      throw ArgumentError("The 'other' case must be specified");
    }
    return other;
  }
```

So I think that failing meaningfully in both cases is acceptable.

@jmagman WDYT

Reply:
I don't have familiarity with this code, @HansMuller may have opinions.

Reply:
Looking ....